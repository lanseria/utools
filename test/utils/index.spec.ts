import { treeCloneById } from "@lanseria/utools";
import { data2 } from "../mock";

describe("Utils 工具函数", () => {
  it("treeCloneById 拷贝并赋值唯一ID", () => {
    const idList = treeCloneById(data2);
    expect(idList).toMatchSnapshot();
  });
});
