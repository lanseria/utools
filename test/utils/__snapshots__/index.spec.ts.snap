// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils 工具函数 treeCloneById 拷贝并赋值唯一ID 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "id": 2,
          "name": "a11",
        },
        Object {
          "id": 3,
          "name": "a12",
        },
      ],
      "id": 1,
      "name": "a1",
    },
    Object {
      "children": Array [
        Object {
          "id": 5,
          "name": "a21",
        },
        Object {
          "id": 6,
          "name": "a22",
        },
      ],
      "id": 4,
      "name": "a2",
    },
    Object {
      "children": Array [
        Object {
          "id": 8,
          "name": "a31",
        },
        Object {
          "id": 9,
          "name": "a32",
        },
      ],
      "id": 7,
      "name": "a3",
    },
  ],
  "id": 0,
  "name": "a",
}
`;
